src/uct/cuda/cuda_ipc/cuda_ipc_cache.c

```diff
@@ -102,7 +103,42 @@ static void uct_cuda_ipc_cache_invalidate_regions(uct_cuda_ipc_cache_t *cache,
               cache->name, from, to);
 }
 
-UCS_PROFILE_FUNC(ucs_status_t, uct_cuda_ipc_cache_map_memhandle,
+ucs_status_t uct_cuda_ipc_unmap_memhandle(void *rem_cache, uintptr_t d_bptr,
+                                          void *mapped_addr, int cache_enabled)
+{
+    uct_cuda_ipc_cache_t *cache = (uct_cuda_ipc_cache_t *) rem_cache;
+    ucs_status_t status         = UCS_OK;
+    ucs_pgt_region_t *pgt_region;
+    uct_cuda_ipc_cache_region_t *region;
+
+    /* use write lock because cache maybe modified */
+    pthread_rwlock_wrlock(&cache->lock);
+    pgt_region = UCS_PROFILE_CALL(ucs_pgtable_lookup, &cache->pgtable, d_bptr);
+    ucs_assert(pgt_region != NULL);
+    region = ucs_derived_of(pgt_region, uct_cuda_ipc_cache_region_t);
+
+    ucs_assert(region->refcount >= 1);
+    ucs_atomic_sub64(&(region->refcount), (uint64_t) 1);
```

Comment:
seems region->refcount is used only under write-lock so it does not have to be atomic

Reply:
@yosefe Fixed this. 

@bureddy Can you review this one more time? Some of the old code is now changed to use write lock because I noticed that cache structure was being updated under read lock.